### ISSUE TITLE
Eclipse attack texts

### ISSUE THREAD

**Issue by FrankSzendzielarz** [ethereum/devp2p#109]
The eclipse attack texts deletes the previous information and references to the "Bound-LogIDforPhyIDSrc" in this paper https://engineering.purdue.edu/~isl/TR-EE-07-13.pdf 

Instead, the text now makes a general suggestion to limit IPs by bucket and table. 

Both approaches have their limitations. In the former case, it is difficult to establish what the correct weighting of per-learned-from-source should be, and with the latter I feel a determined attacker may still be able to take advantage of delays in the node verification schedule to propagate junk around a target and with self-propagation.

I would prefer if the implementation recommendation was more agnostic about how these attacks should be mitigated, while proposing multiple strategies and referring to the papers.

**Comment by fjl** [comment]
One countermeasure often recommended in literature is proof-of-work. We have a separate issue for that #122.

**Comment by fjl** [comment]
Just found another very interesting approach, described in [this paper](https://hal.inria.fr/inria-00490509/file/HotP2P10-KAD_DHT_attack_mitigation-cholez.pdf) (also see [slides](http://www.ipdps.org/ipdps2010/ipdps2010-slides/HOTP2P/HotP2P_DHT_attack_mitigation_Cholez.pdf) for an overview.

The idea is simple: during lookup activity, track node ID distribution. Since node IDs are uniformly distributed across the entire ID number space, any deviations from the regular distributin are malicious and those nodes should be ignored.

**Comment by fjl** [comment]
I still want to add the above thing to the spec and try it out in implementation.

**Comment by contrun** [comment]
> Just found another very interesting approach, described in [this paper](https://hal.inria.fr/inria-00490509/file/HotP2P10-KAD_DHT_attack_mitigation-cholez.pdf) (also see [slides](http://www.ipdps.org/ipdps2010/ipdps2010-slides/HOTP2P/HotP2P_DHT_attack_mitigation_Cholez.pdf) for an overview.
> 
> The idea is simple: during lookup activity, track node ID distribution. Since node IDs are uniformly distributed across the entire ID number space, any deviations from the regular distribution are malicious and those nodes should be ignored.
> 
> Found this while browsing [gtk-gnutella source code](https://github.com/gtk-gnutella/gtk-gnutella/blob/44e87acf49adcf30affc654c95bdd9f97db40540/src/dht/lookup.c#L107).

I don't think this countermeasure is effective in discv5. This countermeasure is needed because attackers can choose whatever node ID closer to the target in the DHT network like emule, while in discv5 the node ID is actually a hash digest. The attacker has no way to effectively choose node ID closer to the targets.

**Comment by kdeme** [comment]
> The attacker has no way to effectively choose node ID closer to the targets.

While more costly, the attacker can still "mine" for those node IDs.

**Comment by contrun** [comment]
> > The attacker has no way to effectively choose node ID closer to the targets.
> 
> While more costly, the attacker can still "mine" for those node IDs.

Fair point.
